### PR TITLE
BAF-939 Integration test failing

### PR DESCRIPTION
**Pull request by MarioIvancik**
Bug report:
https://youtrack.bringauto.com/issue/BAF-939/Module-Gateway-tests-are-not-passing

## Summary by CodeRabbit

- **New Features**
  - Updated the version of the Module Gateway, indicating added enhancements and improvements.
  - Introduced validation in command handling to prevent processing of empty command data, improving error handling.
  - Simplified buffer handling and initialization across various modules for enhanced performance.
  - Enhanced constructors in the InternalClientMessage class for better resource management.
  - Improved logging clarity in the MqttCommunication class.
  - Updated constant types for better clarity and maintainability.

- **Bug Fixes**
  - Enhanced error logging and management for several operations, ensuring better tracking of issues.

### CODE REVIEW

File: source/bringauto/modules/StatusAggregator.cpp [hunk 1]
```diff
@@ -27,7 +27,10 @@ StatusAggregator::aggregateStatus(structures::StatusAggregatorDeviceState &devic
 								  const unsigned int &device_type) {
 	auto &currStatus = deviceState.getStatus();
 	Buffer aggregatedStatusBuff {};
-	module_->aggregateStatus(aggregatedStatusBuff, currStatus, status, device_type);
+	if (module_->aggregateStatus(aggregatedStatusBuff, currStatus, status, device_type) != OK) {
+		log::logWarning("Error occurred while aggregating status, returning empty buffer");
+		aggregatedStatusBuff = module_->constructBuffer();
```
Comment (koudis) [inline source/bringauto/modules/StatusAggregator.cpp:32]: What if the aggregatedStatusBuff is already allocated?
Ha, then the Buffer is deallocated and replaced by a new buffer!

But when the constructBUffer is needed here?

Comment (MarioIvancik) [reply source/bringauto/modules/StatusAggregator.cpp]: after discussion, it was decided that module functions will construct an empty buffer if they fail

File: source/bringauto/modules/StatusAggregator.cpp [hunk 2]
```diff
@@ -27,7 +27,9 @@ StatusAggregator::aggregateStatus(structures::StatusAggregatorDeviceState &devic
 								  const unsigned int &device_type) {
 	auto &currStatus = deviceState.getStatus();
 	Buffer aggregatedStatusBuff {};
-	module_->aggregateStatus(aggregatedStatusBuff, currStatus, status, device_type);
+	if (module_->aggregateStatus(aggregatedStatusBuff, currStatus, status, device_type) != OK) {
```
Comment (koudis) [inline source/bringauto/modules/StatusAggregator.cpp:30]: if aggregate status fails it sends an empty status to the external server - is this behaviour correct?

Comment (MarioIvancik) [reply source/bringauto/modules/StatusAggregator.cpp:30]: aggregateStatus will now return the current status if the module function fails

**Review by koudis — APPROVED** [review]

**Pull request merged**